src/main/java/org/embulk/filter/encrypt/EncryptFilterPlugin.java

```diff
@@ -178,35 +247,138 @@ public void transaction(ConfigSource config, Schema inputSchema,
     {
         PluginTask task = config.loadConfig(PluginTask.class);
 
-        if (task.getAlgorithm().useIv() && !task.getIvHex().isPresent()) {
-            throw new ConfigException("Algorithm '" + task.getAlgorithm() + "' requires initialization vector. Please generate one and set it to iv_hex option.");
+        validateAndResolveKey(task, inputSchema);
+
+        control.run(task.dump(), inputSchema);
+    }
+
+    @VisibleForTesting
+    public Map<String, String> retrieveKey(final String bucket, final String path, final AmazonS3 client)
+    {
+        S3Object fullObject = null;
+
+        try {
+            fullObject = client.getObject(new GetObjectRequest(bucket, path));
+            return (Map<String, String>) yaml.load(fullObject.getObjectContent());
```

Comment:
I think it will be a ClassCastException here if yaml load any other types (array, just string,...). So, same as above, we should probably throwing a more meaningful exception.

Reply:
Fixed as same above.